samples/apps/copilot-chat-app/webapi/Controllers/BotController.cs

```diff
@@ -0,0 +1,254 @@
+    public async Task<ActionResult> UploadAsync(
+        [FromServices] Kernel kernel,
+        [FromQuery] string userId,
+        [FromBody] Bot bot)
+    {
+        // TODO: We should get userId from server context instead of from request for privacy/security reasons when support multipe users.
+        this._logger.LogDebug("Received call to upload a bot");
+
+        if (!this.IsBotCompatible(bot.Schema, bot.EmbeddingConfigurations))
+        {
+            return new UnsupportedMediaTypeResult();
+        }
+
+        string chatTitle = $"{bot.ChatTitle} - Clone";
+        string chatId = string.Empty;
+
+        // Upload chat history into chat repository and embeddings into memory.
+        try
+        {
+            // 1. Create a new chat and get the chat id.
+            var newChat = new ChatSession(userId, chatTitle);
+            await this._chatRepository.CreateAsync(newChat);
+            chatId = newChat.Id;
+
+            string oldChatId = bot.ChatHistory.First().ChatId;
+
+            // 2. Update the app's chat storage.
+            foreach (var message in bot.ChatHistory)
+            {
+                var chatMessage = new ChatMessage(message.UserId, message.UserName, chatId, message.Content, ChatMessage.AuthorRoles.Participant);
+                chatMessage.Timestamp = message.Timestamp;
+                await this._chatMessageRepository.CreateAsync(chatMessage);
+            }
+
+            // 3. Update the memory.
+            await this.BulkUpsertMemoryRecordsAsync(oldChatId, chatId, bot.Embeddings);
+        }
+        catch
+        {
+            // TODO: Revert changes if any of the actions failed
+            throw;
+        }
+
+        return this.Accepted();
+    }
+
+    /// <summary>
+    /// Download a bot.
+    /// </summary>
+    /// <param name="kernel"></param>
+    /// <param name="chatId"></param>
+    /// <returns></returns>
+    [Authorize]
+    [HttpGet]
+    [Route("bot/download/{chatId:guid}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<ActionResult<string>> DownloadAsync(
+        [FromServices] Kernel kernel,
+        Guid chatId)
+    {
+        this._logger.LogDebug("Received call to download a bot");
+        var memory = await this.CreateBotAsync(kernel, this._chatRepository, this._chatMessageRepository, chatId);
+
+        return JsonSerializer.Serialize(memory);
+    }
+
+    /// <summary>
+    /// Check if an external bot file is compatible with the application.
+    /// </summary>
+    /// <remarks>
+    /// If the embeddings are not generated from the same model, the bot file is not compatible.
+    /// </remarks>
+    /// <param name="externalBotSchema">The external bot schemal.</param>
+    /// <param name="externalBotEmbeddingConfig">The external bot embedding configuration.</param>
+    /// <returns></returns>
```

Comment:
[nit] Populate or remove empty XMLDocs

Reply:
good catch. Do you have a tool to spot missing XMLDocs? If so, I would love to setup one in my dev env.